lib/parslet/atoms/sequence.rb

```diff
@@ -29,9 +29,10 @@ def try(source, context, consume_all)
       success, value = p.apply(source, context, child_consume_all) 
 
       unless success
-        return context.err(self, source, @error_msgs[:failed], [value]) 
+        return context.err(self, source, @error_msgs[:failed], [value])
       end
-      
+      context.succ(source)
```

Comment:
This is indeed a delicate area :) 
At the end of the day there seems to be two fundamental parsing expressions for a PEG as far as error reporting goes: the sequence and the ordered choice (alternative in parslet parlance). The idea behind the contextual reporter is that it's probably not useful to report an error that happened on an expression that was part of an ordered choice for which another expression successfully parsed. However if the expression that did parse successfully was not a sequence then it is more likely that the previous error is still relevant. Maybe a clearer way to say the same thing is: if a sequence expression that was a choice in an ordered choice expression parsed successfully then any previous error that occurred while parsing expressions part of the ordered choice expression is most likely irrelevant to the user (because matching a sequence is more "difficult"). 
This is a heuristic obviously so it may be that reporting success for all expressions may end up creating better error messages in some cases.